contracts/src/bridge/SequencerInbox.sol

```diff
@@ -133,6 +136,13 @@ contract SequencerInbox is DelegateCallAware, GasRefundEnabled, ISequencerInbox
         if (msg.sender != tx.origin) revert NotOrigin();
         if (!isBatchPoster[msg.sender]) revert NotBatchPoster();
         if (inboxAccs.length != sequenceNumber) revert BadSequencerNumber();
+        bytes32 dasKeysetHash = dasKeysetHashFromBatchData(data);
+        if (dasKeysetHash != bytes32(0)) {
+            if (
+                dasKeysetManager == DasKeysetManager(address(0)) ||
+                !dasKeysetManager.isValidKeysetHash(dasKeysetHash)
+            ) revert InvalidDASKeyset(dasKeysetHash);
+        }
```

Comment:
This code should be in `formDataHash` (currently it's not called for `addSequencerL2Batch`, the non-origin variant)